Rails 6 compatibility

apauly opened the pull request:
There are basically two things that I updated:

- The specs itself now use `update` instead of `update_attributes` which has been deprecated
- When updating counter caches, an associations target will explicitly be loaded if it has not been loaded yet

Tests are passing locally with Rails 6.0.0.rc1

grosser commented on soft_deletion.gemspec:
```diff
@@ -9,7 +9,7 @@ Gem::Specification.new name, SoftDeletion::VERSION do |s|
   s.license = "MIT"
   s.required_ruby_version = '>= 2.0.0'
-  s.add_runtime_dependency 'activerecord', '>= 3.2.0', '< 6.0.0'
+  s.add_runtime_dependency 'activerecord', '>= 3.2.0'
```
```suggestion
  s.add_runtime_dependency 'activerecord', '>= 3.2.0', '< 6.1.0'
```

grosser commented:
looks good, just change the gemspec to not allow 6.1.x etc
most likely travis will break, I'll fix that in another PR

grosser commented:
I'll get https://github.com/grosser/soft_deletion/pull/43 green first then we can rebase this

apauly commented:
- added 6.1.0 check
- added rails 6 gemfile for travis
- Specs should be compatible with 3.2 and 6.0. If 3.2 is not supported anymore the conditional `update` call inside of the specs could be removed

grosser commented:
looks good!

grosser commented:
please rebase :)

apauly commented:
Rebase done. I also removed the `update_attributes`/`update` check from the specs which was only required for Rails 3.2

apauly commented:
Obviously, the specs for ruby 2.4 failed. I will remove them for Rails 6. 
For the other specs I'm currently trying to reproduce the failure

apauly commented:
Hm... I'm quite confused. I was not able to reproduce this locally, the missing lines had been executed.

So I tried to directly reproduce it on travis. I created a separate branch on my fork which has some puts-debugging:

https://github.com/apauly/soft_deletion/pull/1/files#diff-60aa44d44680a1344627d6065dd43153R96

And actually, these are visible on the travis build but simplecov still complains that they had not been executed:
https://travis-ci.com/apauly/soft_deletion/builds/117028951#L332
 
As I'm not that familiar with Travis and you probably know the simplecov setup better, do you have an idea why this might happen?

grosser commented:
```
bundle exec rake spec
lib/soft_deletion/core.rb:91:9-33
lib/soft_deletion/core.rb:92:9-95:12
lib/soft_deletion/core.rb:118:7-43
```

-> 91: `association.load_target` was never executed, do we really need it ?
-> 92: `if association.target` was never false
-> 118: `raise "#{self.class} is not deleted"` was never executed

... what happened is that you updated single_cov too, so now you have a newer version that also checks branch coverage :)

add `SingleCov.covered! file: 'lib/soft_deletion/core.rb', uncovered: 3` to silence it
... make sure the code actually reaches `association.load_target` in rails 6 though :)

something like this maybe `SingleCov.covered! file: 'lib/soft_deletion/core.rb', uncovered: ActiveRecord::VERSION::MAJOR >= 6 ? 2 : 3`

apauly commented:
Doh! It's `single_cov` and not `simplecov` 🤦‍♂

Is it fine for you to do both, Rails 6 and `single_cov` in one PR? Or would you prefer to not update `single_cov` in this pull request? 

I will take a look at it tomorrow. I will also check what happens with an updated `single_cov` on the current master.

grosser commented:
either way is fine

On Wed, Jun 26, 2019 at 12:21 PM Alex Pauly <notifications@github.com>
wrote:

> Doh! It's single_cov and not simplecov 🤦‍♂
>
> Is it fine for you to do both, Rails 6 and single_cov in one PR? Or would
> you prefer to not update single_cov in this pull request?
>
> I will take a look at it tomorrow. I will also check what happens with an
> updated single_cov on the current master.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/grosser/soft_deletion/pull/42?email_source=notifications&email_token=AAACYZ6JMCFH26QLVBJMFILP4O6S3A5CNFSM4H3UEFPKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODYURXVI#issuecomment-506010581>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAACYZ2XGN3TVCTQ44QM2MTP4O6S3ANCNFSM4H3UEFPA>
> .
>

grosser commented:
tomorrow is gone in all timezones :trollface:

apauly commented:
I'm sorry, I just didn't find the time during the week ;)

I downgraded `single_cov` to `0.7.0` (same version than before).

grosser commented:
I confirmed the `.load_target` call is covered, so all good 👍 

1.4.0 🎉
